### PR TITLE
Unbind containers if they are no longer running

### LINKED ISSUE #1749: network ls crashes docker personality after running container
The sequence of commands:

```
docker run -it busybox
docker network ls
```
crashes docker personality.

### PR DESCRIPTION
**Pull request by hmahmood**
Fixes #1749

### CODE REVIEW

File: lib/portlayer/exec/container.go [hunk 1]
```diff
@@ -280,69 +301,65 @@ func (c *Container) Remove(ctx context.Context, sess *session.Session) error {
...
-		return nil, fmt.Errorf("%s does not appear to be a container", containerID)
+		return fmt.Errorf("not a container")
 	case 1:
 		// we have a winner
-		return cc[0], nil
-	default:
-		// we manged to find multiple vms
-		return nil, fmt.Errorf("multiple containers named %s found", containerID)
+		cc[0].vm = c.vm
+		c.swap(cc[0])
```
Comment (emlin) [inline lib/portlayer/exec/container.go]: why do you need this swap? Is this to update the cached container information based on VM information?

Comment (hmahmood) [inline lib/portlayer/exec/container.go]: Yes; we update the config and the state, as well as the vm moref.

File: lib/apiservers/portlayer/restapi/handlers/containers_handlers.go [hunk 2]
```diff
@@ -261,7 +264,8 @@ func (handler *ContainersHandlersImpl) GetContainerListHandler(params containers
 		info.ContainerID = &container.ExecConfig.ID
 		info.LayerID = &container.ExecConfig.LayerID
 		info.Created = &container.ExecConfig.Created
-		info.Status = &container.Status
+		info.Status = new(string)
+		*info.Status = container.State.String()
```
Comment (hickeng) [inline lib/apiservers/portlayer/restapi/handlers/containers_handlers.go]: Strings are immutable so:
```
info.Status = &container.State.String()
```

Comment (dougm) [inline lib/apiservers/portlayer/restapi/handlers/containers_handlers.go]: @hickeng Don't think this needs to change, it is consistent with our existing pattern where we deal with 'string *' fields.  And `... = &container.State.String()` won't compile.

**Comment by dougm** [comment]
lgtm

File: lib/apiservers/portlayer/restapi/handlers/containers_handlers.go (hunk 2 above)
Comment (hickeng) [inline lib/apiservers/portlayer/restapi/handlers/containers_handlers.go]: Sorry - apparently I'm idiotic today.

**Comment by caglar10ur** [comment]
LGTM :sheep: :it:

**Comment by hmahmood** [comment]
This does not really fix the problem as further testing has revealed. This will have to be fixed after events are implemented in the port layer.

**Pull request closed without merging**